modules/gapi/src/backends/cpu/gparsers.cpp

```diff
@@ -0,0 +1,332 @@
+        std::tie(rc, image_id, confidence, label) = parser.unitedParse(i);
+
+        if (image_id < 0.f)
+        {
+            break;    // marks end-of-detections
+        }
+
+        if (confidence < confidence_threshold &&
+            filter_label != -1 &&
+            label != filter_label)
+        {
+            continue; // filter out object classes if filter is specified
+        }             // and skip objects with low confidence
+        out_boxes.emplace_back(rc & parser.getSurface());
+        out_labels.emplace_back(label);
+    }
+}
+
+void parseSSD(const cv::Mat&  in_ssd_result,
+              const cv::Size& in_size,
+              const float&    confidence_threshold,
+              const bool&     filter_out_of_bounds,
+              std::vector<cv::Rect>& out_boxes)
+{
+    cv::gapi::parsers::SSDParser parser(in_ssd_result.size, in_size, in_ssd_result.ptr<float>());
+    out_boxes.clear();
+    cv::Rect rc;
+    float image_id, confidence;
+    int label;
+    for (int i = 0; i < parser.getMaxProposals(); ++i)
+    {
+        std::tie(rc, image_id, confidence, label) = parser.unitedParse(i);
+
+        if (image_id < 0.f)
+        {
+            break;    // marks end-of-detections
+        }
+        if (confidence < confidence_threshold)
+        {
+            continue; // skip objects with low confidence
+        }
+
+        parser.adjustBoundingBox(rc);
+
+        const auto clipped_rc = rc & parser.getSurface();
+        if (filter_out_of_bounds)
+        {
+            if (clipped_rc.area() != rc.area())
+            {
+                continue;
+            }
+        }
+        out_boxes.emplace_back(clipped_rc);
+    }
+}
+
+void parseYolo(const cv::Mat&  in_yolo_result,
+               const cv::Size& in_size,
+               const float&    confidence_threshold,
+               const float&    nms_threshold,
+               const cv::gapi::core::YoloAnchors& anchors,
+               std::vector<cv::Rect>& out_boxes,
+               std::vector<int>&      out_labels)
+{
+    const auto& dims = in_yolo_result.size;
+    GAPI_Assert(dims.dims() == 4);
+    GAPI_Assert(dims[0] == 1);
+    GAPI_Assert(dims[1] == 13);
+    GAPI_Assert(dims[2] == 13);
+    GAPI_Assert(dims[3] % 5 == 0); // 5 boxes
+    const auto num_classes = dims[3] / 5-5;
+    GAPI_Assert(num_classes > 0);
+    GAPI_Assert(0 < nms_threshold && nms_threshold <= 1);
+    out_boxes.clear();
+    out_labels.clear();
+    gapi::parsers::YoloParams params;
+    constexpr auto side = 13;
+    constexpr auto side_square = side * side;
+    const auto output = in_yolo_result.ptr<float>();
+
+    gapi::parsers::YoloParser parser(output, side, params.coords, num_classes);
+
+    struct Detection
+    {
+        Detection(const cv::Rect& in_rect, const float& in_conf, const int& in_label)
+            : rect(in_rect), conf(in_conf), label(in_label)
+        {}
+        cv::Rect rect;
+        float    conf = 0.0f;
+        int      label = 0;
+    };
```

Comment:
Shouldn't it be defined in `YoloParser` class?

Reply:
or some other place out of the function

Reply:
Fixed